ta/avb/entry.c

```diff
@@ -0,0 +1,258 @@
+static TEE_Result write_rb_idx(uint32_t pt, TEE_Param params[TEE_NUM_PARAMS])
+{
+	const uint32_t exp_pt = TEE_PARAM_TYPES(TEE_PARAM_TYPE_VALUE_INPUT,
+						TEE_PARAM_TYPE_VALUE_INPUT,
+						TEE_PARAM_TYPE_NONE,
+						TEE_PARAM_TYPE_NONE);
+	size_t slot_offset;
+	uint64_t widx;
+	uint64_t idx;
+	uint32_t count;
+	TEE_Result res;
+	TEE_ObjectHandle h;
+
+	if (pt != exp_pt)
+		return TEE_ERROR_BAD_PARAMETERS;
+
+	res = get_slot_offset(params[0].value.a, &slot_offset);
+	if (res)
+		return res;
+	widx = ((uint64_t)params[1].value.a << 32) | params[1].value.b;
+
+	res = open_state(DEFAULT_LOCK_STATE, &h);
+	if (res)
+		return res;
+
+	res = TEE_SeekObjectData(h, slot_offset, TEE_DATA_SEEK_SET);
+	if (res)
+		goto out;
+
+	res =  TEE_ReadObjectData(h, &idx, sizeof(idx), &count);
+	if (res)
+		goto out;
+	if (count != sizeof(idx))
+		idx = 0; /* Not yet written slots are reported as 0 */
+
+	if (widx < idx) {
+		res = TEE_ERROR_SECURITY;
+		goto out;
+	}
+
+	res = TEE_SeekObjectData(h, slot_offset, TEE_DATA_SEEK_SET);
+	if (res)
+		goto out;
+
+	res = TEE_WriteObjectData(h, &widx, sizeof(widx));
+out:
+	TEE_CloseObject(h);
+	return res;
+}
+
+static TEE_Result read_lock_state(uint32_t pt, TEE_Param params[TEE_NUM_PARAMS])
+{
+	const uint32_t exp_pt = TEE_PARAM_TYPES(TEE_PARAM_TYPE_VALUE_OUTPUT,
+						TEE_PARAM_TYPE_NONE,
+						TEE_PARAM_TYPE_NONE,
+						TEE_PARAM_TYPE_NONE);
+	uint32_t lock_state;
+	uint32_t count;
+	TEE_Result res;
+	TEE_ObjectHandle h;
+
+	if (pt != exp_pt)
+		return TEE_ERROR_BAD_PARAMETERS;
+
+	res = open_state(DEFAULT_LOCK_STATE, &h);
+	if (res)
+		return res;
+
+	res =  TEE_ReadObjectData(h, &lock_state, sizeof(lock_state), &count);
+	if (res)
+		goto out;
+	if (count != sizeof(lock_state)) {
+		/*
+		 * Client need write the lock state to recover, this can
+		 * normally not happen.
+		 */
+		res = TEE_ERROR_CORRUPT_OBJECT;
+		goto out;
+	}
+
+	params[0].value.a = lock_state;
+out:
+	TEE_CloseObject(h);
+	return res;
+}
+
+static TEE_Result write_lock_state(uint32_t pt,
+				   TEE_Param params[TEE_NUM_PARAMS])
+{
+	const uint32_t exp_pt = TEE_PARAM_TYPES(TEE_PARAM_TYPE_VALUE_INPUT,
+						TEE_PARAM_TYPE_NONE,
+						TEE_PARAM_TYPE_NONE,
+						TEE_PARAM_TYPE_NONE);
+	uint32_t wlock_state;
+	uint32_t lock_state;
+	uint32_t count;
+	TEE_Result res;
+	TEE_ObjectHandle h;
+
+	if (pt != exp_pt)
+		return TEE_ERROR_BAD_PARAMETERS;
+
+	wlock_state = params[0].value.a;
+
+	res = open_state(wlock_state, &h);
+	if (res)
+		return res;
+
+	res =  TEE_ReadObjectData(h, &lock_state, sizeof(lock_state), &count);
+	if (res)
+		goto out;
+	if (count == sizeof(lock_state) && lock_state == wlock_state)
+		goto out;
+
+	res = create_state(wlock_state, &h);
+out:
+	TEE_CloseObject(h);
+	return res;
+}
+
+TEE_Result TA_CreateEntryPoint(void)
+{
+	return TEE_SUCCESS;
+}
+
+void TA_DestroyEntryPoint(void)
+{
+}
+
+TEE_Result TA_OpenSessionEntryPoint(uint32_t __unused pt,
+				    TEE_Param __unused params[4],
+				    void __unused **session)
```

Comment:
Move `__unused` after the parameter name, it will make checkpatch happy I suppose. It's what we have in all other optee_os files.

Reply:
I'll fix